ansible/initialize-master.yml

```diff
@@ -56,10 +56,32 @@
       until: result is not failed
       when: initialized is changed
 
+      # Azure doesn't support IP-in-IP so run calico in VXLAN mode
+    - name: Set calico to use VXLAN instead of IP-in-IP
+      command: kubectl --kubeconfig /etc/kubernetes/admin.conf patch daemonsets. -n kube-system calico-node --patch "{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"name\":\"calico-node\",\"env\":[{\"name\":\"CALICO_IPV4POOL_IPIP\",\"value\":\"Never\"},{\"name\":\"CALICO_IPV4POOL_VXLAN\",\"value\":\"Always\"}]}]}}}}"
+      when: cloud_provider is defined and cloud_provider == "azure"
+
     - name: Force calico to use correct interface
       command: kubectl --kubeconfig /etc/kubernetes/admin.conf patch daemonsets. -n kube-system calico-node --patch "{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"name\":\"calico-node\",\"env\":[{\"name\":\"IP_AUTODETECTION_METHOD\",\"value\":\"can-reach={{ private_network_cidr | ipaddr(-2) | ipaddr('address') }}\"}]}]}}}}"
       when: private_network_cidr is defined
 
+    - name: Render azure cloud controller
+      template:
+        src: azure-cloud-controller.yaml.j2
+        dest: /etc/kubeadm/azure-cloud-controller.yaml
+        owner: root
+        group: root
+        mode: '0644'
+      when: cloud_provider is defined and cloud_provider == "azure"
+
+    - name: Install azure cloud controller
+      command: kubectl --kubeconfig /etc/kubernetes/admin.conf apply -f /etc/kubeadm/azure-cloud-controller.yaml
+      register: result
+      retries: 15
+      delay: 5
```

Comment:
@viktor-f Do you know if you added this for a reason?

Reply:
The Apply calico had it so I just copied it. But I'm not sure if it's actually needed. If it is needed I agree that it should be added to the patches as well.

Reply:
It could be that it's only the first task that needs it since the API server might not start up in time. But it shouldn't hurt retrying `kubectl apply` anyway so I would be fine adding it to all of those.

On the other hand, if we start getting a lot of kubectl patches and commands perhaps it would make sense to manage them differently. For example patches would make sense to vendor and template instead of patch probably.